packages/ai-bot/lib/send-response.ts

```diff
@@ -81,38 +87,47 @@ export class Responder {
 
   async onMessage(msg: {
     role: string;
-    tool_calls?: { function: { name: string; arguments: string } }[];
+    tool_calls?: { function: FunctionToolCall }[];
   }) {
     if (msg.role === 'assistant') {
-      for (const toolCall of msg.tool_calls || []) {
-        const functionCall = toolCall.function;
-        log.debug('[Room Timeline] Function call', toolCall);
-        let args;
-        try {
-          args = JSON.parse(functionCall.arguments);
-        } catch (error) {
-          Sentry.captureException(error);
-          this.initialMessageReplaced = true;
-          let errorPromise = sendError(
-            this.client,
-            this.roomId,
-            error,
-            this.initialMessageReplaced ? undefined : this.initialMessageId,
-          );
-          this.messagePromises.push(errorPromise);
-          await errorPromise;
-        }
-        if (functionCall.name === 'patchCard') {
+      await this.handleFunctionToolCalls(msg);
+    }
+  }
+
+  async handleFunctionToolCalls(msg: {
+    role: string;
+    tool_calls?: { function: FunctionToolCall }[];
+  }) {
+    for (const toolCall of msg.tool_calls || []) {
+      const functionCall = toolCall.function;
+      log.debug('[Room Timeline] Function call', toolCall);
+      try {
+        functionCall.arguments = JSON.parse(functionCall.arguments);
+        if (
+          functionCall.name === 'patchCard' ||
+          functionCall.name === 'searchCard'
```

Comment:
also it is sad that we have specific command logic growing in the ai-bot. this is not sustainable, we need to think about solutions that do not require ai-bot changes when we add new commands.